sonar-batch/src/main/java/org/sonar/batch/bootstrap/PersistentCacheProvider.java

```diff
@@ -36,9 +42,22 @@ public PersistentCache provide(UserProperties props) {
         builder.setSonarHome(Paths.get(home));
       }
 
+      builder.setVersion(getVersion());
       cache = builder.build();
-    } 
-    
+    }
+
     return cache;
   }
+
+  private String getVersion() {
+    InputStream is = this.getClass().getClassLoader().getResourceAsStream("sq-version.txt");
+    if (is == null) {
+      return null;
+    }
+    try (BufferedReader br = new BufferedReader(new InputStreamReader(is, StandardCharsets.UTF_8))) {
+      return br.readLine();
+    } catch (IOException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)